backend/tracim_backend/lib/utils/translation.py

```diff
@@ -21,33 +26,72 @@ def translator_marker(string: str) -> str:
     return string
 
 
+class TranslationSource(Enum):
+    GLOBAL = "*/backend.json"
+    CUSTOM_PROPERTIES = "*.json"
+
+
 class Translator(object):
     """
     Get translation from json file
     """
 
-    def __init__(self, app_config: "CFG", default_lang: str = None, fallback_lang: str = None):
+    def __init__(
+        self,
+        app_config: "CFG",
+        default_lang: str = None,
+        fallback_lang: str = None,
+        default_source: TranslationSource = TranslationSource.GLOBAL,
+    ):
         """
         you should provide either valid fallback_lang(true value) or valid
         app.config.DEFAULT_LANG (true value).
         """
-        self.config = app_config
         if not fallback_lang:
             assert app_config.DEFAULT_LANG
             fallback_lang = app_config.DEFAULT_LANG
         self.fallback_lang = fallback_lang
         if not default_lang:
             default_lang = fallback_lang
         self.default_lang = default_lang
+        self.default_source = default_source
+        self.app_config = app_config
 
-    def _get_json_translation_lang_filepath(self, lang: str) -> typing.Optional[str]:
-        i18n_folder = self.config.BACKEND__I18N_FOLDER_PATH
-        lang_filepath = os.path.join(i18n_folder, lang, TRANSLATION_FILENAME)
-        if not os.path.isdir(self.config.BACKEND__I18N_FOLDER_PATH):
-            return None
-        else:
-            return lang_filepath
+    @classmethod
+    def init_translations(cls, app_config: "CFG") -> "CFG":
+        app_config.TRANSLATIONS = {
+            TranslationSource.GLOBAL.name: {},
+            TranslationSource.CUSTOM_PROPERTIES.name: {},
+        }
+        # global translations
+        files = glob.glob(
+            str(pathlib.Path(app_config.BACKEND__I18N_FOLDER_PATH, TranslationSource.GLOBAL.value))
+        )
+        for file in files:
+            lang = basename(dirname(file))
+            if is_file_readable(file):
+                app_config.TRANSLATIONS[TranslationSource.GLOBAL.name][
+                    lang
+                ] = cls._get_translation_from_file(file)
+
+        # custom_properties translations
+        if app_config.USER__CUSTOM_PROPERTIES__TRANSLATIONS_DIR_PATH:
+            files = glob.glob(
+                str(
+                    pathlib.Path(
+                        app_config.USER__CUSTOM_PROPERTIES__TRANSLATIONS_DIR_PATH,
+                        TranslationSource.CUSTOM_PROPERTIES.value,
+                    )
+                )
+            )
+            for file in files:
+                lang = basename(file).split(".")[0]
+                if is_file_readable(file):
```

Comment:
Aren't you hiding possible errors here? Shouldn't the behavior be to fail if a file is not readable?

Reply:
Right. It's really a special case, as this appear at translation loading. This is currently loaded by the config at the end of the config, automatically (don't think it's necessary to separate more for the moment). We should probably add specific error here.

Reply:
done.